[MIG] website_sale_secondary_unit : Migration to 14.0

kimirondon opened the pull request.

sergio-teruel committed:
[11.0][ADD] website_sale_secondary_unit: New module

sergio-teruel committed:
[11.0][FIX] website_sale_secondary_unit: Crash if product_uom not come in values

sergio-teruel committed:
[11.0][FIX] website_sale_secondary_unit: Fix tour tests

sergio-teruel committed:
[11.0][IMP] website_sale_secondary_unit: Display product uom after price when product has secondary units defined

sergio-teruel committed:
[MIG] website_sale_secondary_unit: Migration to v12.0

sergio-teruel committed:
[FIX] website_sale_secondary_unit: Fix quantity position in cart lines

sergio-teruel committed:
[IMP] website_sale_secondary_unit: Set default secondary uom for products when "Add to cart" view is enabled.

AndreVolksdorf committed:
Added translation using Weblate (German)

CarlosRoca13 committed:
[IMP] website_sale_secondary_unit: black, isort, prettier

CarlosRoca13 committed:
[MIG] website_sale_secondary_unit: Migration to v13.0

oca-travis committed:
[UPD] Update website_sale_secondary_unit.pot

OCA-git-bot committed:
[UPD] README.rst

kimirondon commented:
depends on this PR https://github.com/OCA/sale-workflow/pull/1428

CarlosRoca13 committed:
[FIX] website_sale_secondary_unit: Fixed price computation

Before this changes, the change in the value of the secondary unit input field
didn't launch the execution of the onchanges of the real input field.

Adding this changes, we achieve to capture the event of the new input field
to launch the onchanges of the real input field.

cc @Tecnativa TT29708

sergio-teruel committed:
[FIX] website_sale_secondary_unit: Remove sudo access to avoid to display unpublished secondary units.
TT31018

sergio-teruel commented:
@kimirondon please can yo add this code from v13.0
* https://github.com/OCA/e-commerce/pull/558

kimirondon commented:
@sergio-teruel done!
the checks are failing because of the dependency with module sale_order_secondary_unit, waiting for this PR https://github.com/OCA/sale-workflow/pull/1671

pedrobaeza commented:
Please include #573

CarlosRoca13 commented on oca_dependencies.txt:
```diff
@@ -1 +1,2 @@
 # See https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#oca_dependencies-txt
+sale-workflow https://github.com/archeti-org/sale-workflow 14.0-mig-sale_order_secondary_unit
```
```suggestion
sale-workflow
```

kimirondon replied:
Hello! 
I updated the oca_dependencies.txt file as you recommend me but still have this issue:
ERROR: Could not find a version that satisfies the requirement odoo14-addon-sale_order_secondary_unit (from versions: none)

CarlosRoca13 reviewed (commented):
I think that this is blocking

CarlosRoca13 committed:
[FIX] website_sale_secondary_unit: add_qty rounding when not integer secondary unit factor

pedrobaeza commented:
Please include #641 and make Travis green.

kimirondon commented:
@pedrobaeza 
hello! it's done I added the commit from #641, the checks fails due to a dependency on sale_order_secondary_unit which hasn't been migrated yet, PR https://github.com/OCA/sale-workflow/pull/1671

CarlosRoca13 committed:
[FIX] website_sale_secondary_unit: Avoid show non published secondary_units

CarlosRoca13 commented:
Please add this PR https://github.com/OCA/e-commerce/pull/653

kimirondon commented:
hello @CarlosRoca13  done!

chienandalu reviewed (commented):
@kimirondon rebase with current `14.0` head to get new CIs

francesco-ooops commented:
@kimirondon can you rebase?

kimirondon commented:
hello @francesco-ooops!  it's done.

francesco-ooops commented:
> hello @francesco-ooops!  it's done.
> 

Thanks! Can you try to fix runboat for functional review?

chienandalu committed:
[IMP] website_sale_secondary_unit: mobile UX

For mobile users adding an extra column in the cart summary makes that
the others content won't fit very well, specially for products with long
names. As we can infer the sale line information without the secondary
units computation computation column we choose to hide it if the screen
is too small as Odoo does with the product image column.

TT37137

chienandalu commented:
Please include this changes: https://github.com/OCA/e-commerce/pull/712

kimirondon commented:
hello @chienandalu ! done I added your commit.

chienandalu reviewed (commented):
Tests are failing

Commit:
[IMP] website_sale_secondary_unit: black, isort, prettier

kimirondon commented:
hello @chienandalu ! I updated and rebase my branch. The tests are failing due a missing field (active) in the model "product.secondary.unit", this field exist in v13 and also in v15 but not in v14. Should I modify my controller or should I wait until this is fixed in v14 ?

https://github.com/OCA/product-attribute/blob/13.0/product_secondary_unit/models/product_second_unit.py#L27
https://github.com/OCA/product-attribute/blob/15.0/product_secondary_unit/models/product_second_unit.py#L45C42-L45C42

chienandalu commented:
Well, it's been a long time since this was opened and several things could change in the meantime :sweat_smile: 

I'd check for missing fw-ports of changes to v14 like the one you spotted in the dependencies of this module first. You can use the great [oca-port](https://github.com/OCA/oca-port) tool for it.

kimirondon commented:
hello @chienandalu ! thanks a lot for the help, I used oca-port as you recommended, it work really nice.
this is the corresponding PR 
https://github.com/OCA/product-attribute/pull/1388

Commit:
[MIG] website_sale_secondary_unit: Migration to 14.0
